xwiki-platform-core/xwiki-platform-test/xwiki-platform-test-ui/src/main/java/org/xwiki/test/ui/po/FormContainerElement.java

```diff
@@ -73,20 +73,24 @@ protected WebElement getFormElement()
 
     public void fillFieldsByName(Map<String, String> valuesByNames)
     {
-        Map<WebElement, String> valuesByElements = new LinkedHashMap<>((int) (valuesByNames.size() / 0.75));
-
+        Map<WebElement, String> valuesByElements = new LinkedHashMap<>();
+        
+        String lastElementName = null;
         for (String name : valuesByNames.keySet()) {
             valuesByElements.put(getFormElement().findElement(By.name(name)), valuesByNames.get(name));
+            lastElementName = name;
```

Comment:
Wouldn't it be simpler to just keep the WebElement?

Reply:
Addressed in https://github.com/xwiki/xwiki-platform/pull/3675/commits/75ca724069ded9bc8d66b4c745ad0cb8beab13c3 :+1: 

I'm not sure how much more simpler it is but it works too.